cypress/support/cypress-commands/lifecycle.js

```diff
@@ -198,13 +198,18 @@ Cypress.Commands.add('validateLifecycleHistoryAndEvents', (state, appId) => {
       const lifecycleEventRequirementId = scenarioRequirement.find((req) =>
         req.hasOwnProperty('event')
       );
+      const appHistoryPrevious = UTILS.getEnvVariable(CONSTANTS.APP_LIFECYCLE_HISTORY);
+      const appHistoryCount = appHistory.length - appHistoryPrevious.length;
 
       // Lifecycle event validation
-      if (lifecycleEventRequirementId && lifecycleEventRequirementId.event) {
-        const appHistoryPrevious = UTILS.getEnvVariable(CONSTANTS.APP_LIFECYCLE_HISTORY);
-        const appHistoryCount = appHistory.length - appHistoryPrevious.length;
+      if (condition == CONSTANTS.STAY) {
+        UTILS.assertWithRequirementLogs(
+          CONSTANTS.LIFECYCLE_NOTIFICATION_NOT_GENERATED,
+          appHistoryCount < 1,
+          true
+        );
+      } else if (lifecycleEventRequirementId && lifecycleEventRequirementId.event) {
         let pretext;
-        // If no lifecycle events expected, validate app history value is also empty
```

Comment:
check TTT on how it is handled
In TTT, we never had this variable IS_SAME_APP_TRANSITION 
TTT used to differentiate between events expected/not expected based on the glue code itself